plugins/modules/backup_vault.py

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from ansible_collections.amazon.aws.plugins.module_utils.tagging import compare_aws_tags
+from ansible_collections.amazon.aws.plugins.module_utils.tagging import boto3_tag_list_to_ansible_dict
+from ansible_collections.amazon.aws.plugins.module_utils.tagging import ansible_dict_to_boto3_tag_list
+from ansible_collections.amazon.aws.plugins.module_utils.modules import AnsibleAWSModule
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+from ansible_collections.amazon.aws.plugins.module_utils.backup import get_backup_resource_tags
+
+
+DOCUMENTATION = r"""
+---
+module: backup_vault
+version_added: 6.0.0
+short_description: manage BackupVault create, delete, list
+description:
+  - Creates, deletes, or lists Backup Vault configuration.
+author:
+  - Gomathi Selvi Srinivasan (@GomathiselviS)
+options:
+  state:
+    description:
+      - Add or remove Backup Vault configuration.
+      - 'The following states have been preserved for backwards compatibility: I(state=enabled) and I(state=disabled).'
+      - I(state=enabled) is equivalet to I(state=present).
+      - I(state=disabled) is equivalet to I(state=absent).
+    type: str
+    choices: ['present', 'absent', 'enabled', 'disabled']
+    default: present
+  backup_vault_name:
+    description:
+      - Name for the Backup Vault.
+      - Names are  unique to the account used to create them and the Amazon Web Services Region where they are created.
+      - They consist of letters, numbers, and hyphens.
+    type: str
+    required: true
+  backup_vault_tags:
```

Comment:
I would probably go with tags rather than backup_vault_tags  instead. @tremble any suggestion?

Reply:
The boto3 API has 'BackupVaultTags'. Hence I added 'tags' as an alias for 'backup_vault_tags'. let me know your views.

Reply:
https://ansible-collections.github.io/amazon.aws/branch/main/collections/amazon/aws/docsite/dev_guidelines.html#dealing-with-tags

Reply:
Unfortunately AWS' APIs are an inconsistent mess.  Some use Tags, some use ResourceTags, some use XYZTags, and they're then not even consistent about list of dicts vs a simple dict.

Our standard is to use "tags", with an alias of "resource_tags".    Using  "backup_vault_tags" just adds to the inconsistency between our modules, which makes it harder for folks to work with them.  It's not so bad for backup vaults since they're simple, but when there's a huge list of parameters, being inconsistent makes it too easy to miss when browsing through the docs.